src/client/components/robot/rate.ts

```diff
@@ -0,0 +1,58 @@
+import { observable } from 'mobx'
+import { runInAction } from 'mobx'
+import { throttle } from 'throttle-debounce'
+
+import { Clock, getClock } from '../../../shared/time/clock'
+
+type RateOpts = {
+  smoothing: number
+  unitTime: number
```

Comment:
What is a unit time?

Reply:
All the internal rate calculcation stuff is in `ms`, but the user may want per second, per minute, etc. This unit time does the conversion. E.g. for `unitTime = 1000`, we get the rate in seconds. It's used when we commit the calculated rate to the public property.

Reply:
Ah right. Any particular reason we perform calculations in milliseconds? Could we prefer seconds? We have tried to use seconds everywhere throughout NUbots and NUsight. You'll note the `BrowserSystemClock` interface deals only in seconds.

Reply:
No specific reason, was incidentally because `(Date|performance).now()` returns milliseconds.

Will change to seconds to match the `Clock` interface.

Reply:
Also probably won't need /minute, /hour, etc. so I'll remove the `unitTime` option.

Reply:
Updated in https://github.com/NUbots/NUsight2/pull/269/commits/d45eebc07e27f8ee6ec63ec12ad609812f2ae61b